ls1intum/Artemis
# [Feature] Instructor dashboards with second correction

### Checklist

### Motivation and Context
With the addition of the second correction feature we want that instructors are able to see each individual assessment. Those should be accessible over the instructor dashboard of the specific exercise type.

### Description
Instructors now can access and conduct individual assessments for each correction round for modeling and text exercises.

### Steps for Testing

**_Edit_: Please double check that all submissions are shown. Those without result, those which are locked and those which already have an assessment.**

1. Test the functionality of the exercise-specific dashboard (see navigation below) in exam mode with two correction rounds. 
Under: exam -> exercise-groups -> submissions for a specific exercise

- Assess an exam for a first time, then move back and check if the lock was created. 
- Continue the first assessment
- Submit the first assessment
- Return to the dashboard and check if you can open the first assessment
- now assess the submission in the second correction round. Check if the feedback from the previous round is loaded.
- If so, move back to the dashboard to check if the assessment is locked. Then continue, submit and open it once again. 
- Assessments of other tutors should also be visible in the dashboard. The latest result's score is visible in the result column, as it is the one that counts for the grade.

2. Test the functionality in exam mode with one correction round
It should only display the buttons for one correction round, and those buttons should work correctly. 

3. Test the functionality not in exam mode, but with a regular exercise (text & modeling) 
It should only display the buttons for one correction round, and those buttons should work correctly. 



### Test Coverage




### Screenshots


![image](https://user-images.githubusercontent.com/33342534/105974342-180c1680-608e-11eb-91d6-d8a2e78e125f.png)
